enumerables.rb

```diff
@@ -0,0 +1,293 @@
+    elsif arg.nil?
+      my_each { |elmt| counter += 1 if elmt != false && !elmt.nil? }
+
+    # My_none? will be used to find out if the collection is not made of a specific object
+    else
+      my_each { |elmt| counter += 1 if elmt == arg }
+    end
+    counter.zero?
+  end
+end
+
+module Enumerable
+  # 7.- ---- my_count Method ----
+
+  def my_count(*arg)
+    count = 0
+
+    if block_given?
+      my_each do |elmt|
+        count += 1 if yield(elmt)
+      end
+    elsif arg.empty?
+      count = size
+    else
+      my_each do |elmt|
+        count += 1 if elmt == arg[0]
+      end
+    end
+    count
+  end
+
+  # 8.- ---- my_map Method ----
+
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+
+    if self.class == Array or self.class == Range
+      my_each { |i| new_arr.push(yield(i)) }
+    elsif self.class == Hash
+      my_each { |key, value| new_arr.push(yield(key, value)) }
+    end
+    new_arr
+  end
+end
+
+module Enumerable
+  # 9.- ---- my_inject Method ----
+
+  def my_inject(*args)
+    memo = to_a[0]
+
+    # Case 1: arguments are given but not a block
+    if !args.empty? && !block_given?
+      # There are an initial value and a symbol which indicates the operation to perform
+      if (args[0].is_a? Numeric or args[0].class == String) && args[1].class == Symbol
+        memo = args[0]
+        if args[1] == :+
+          my_each { |elmt| memo += elmt }
+        elsif args[1] == :-
+          my_each { |elmt| memo -= elmt }
+        elsif args[1] == :*
+          my_each { |elmt| memo *= elmt }
+        elsif args[1] == :/
+          my_each { |elmt| memo /= elmt }
+        elsif args[1] == :**
+          my_each { |elmt| memo **= elmt }
+        elsif args[1] == :%
+          my_each { |elmt| memo %= elmt }
+        end
+
+      elsif args.size == 1 && args[0].class == Symbol # There's only the operation's symbol
+        if args[0] == :+
+          self[1..-1].my_each { |elmt| memo += elmt }
+        elsif args[0] == :-
+          self[1..-1].my_each { |elmt| memo -= elmt }
+        elsif args[0] == :*
+          self[1..-1].my_each { |elmt| memo *= elmt }
+        elsif args[0] == :/
+          self[1..-1].my_each { |elmt| memo /= elmt }
+        elsif args[0] == :**
+          self[1..-1].my_each { |elmt| memo **= elmt }
+        elsif args[0] == :%
+          self[1..-1].my_each { |elmt| memo %= elmt }
+        end
+
+      end
+
+      # Case 2: an initial value is given as an argument along with a block
+    elsif (args[0].is_a? Numeric or args[0].class == String) && block_given?
+      memo = args[0]
+      my_each { |elmt| memo = yield(memo, elmt) }
+
+      # Case 3: only a block is given
+    elsif args.empty? && block_given?
+      self[1..-1].my_each { |elmt| memo = yield(memo, elmt) }
+    end
+
+    memo
+  end
+
+  # 10.- ---- Method created for testing my_inject method ----
+
+  def multiply_els(array)
+    array.my_inject(:*)
+  end
```

Comment:
- [ ] enumerables #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject
<img width="671" alt="0" src="https://user-images.githubusercontent.com/58504655/98773216-47ee3b00-240a-11eb-9d78-08c216222c91.PNG">

Reply:
Thank you for the observation. I had a look at this method, made few changes, and tested on Repl.it. Now is returning the result of a multiplication 
![image](https://user-images.githubusercontent.com/63531796/98876797-a33d1f00-2444-11eb-82ef-49ea3ea17fa8.png)